### ISSUE TITLE
Inference  on cpu

### ISSUE THREAD

**Issue by robmarkcole** [worldstrat/worldstrat#3]
In the `Inference.ipynb` notebook cuda is assumed, preventing running on a CPU machine by use of `to("cuda")`

This could be addressed by using a variable `device = "cpu" # or "cuda"` then using `to(device)`

However I still get error `AssertionError: Torch not compiled with CUDA enabled` which means `cudatoolkit` is required in the requirements I believe

**Comment by ivanorsolic** [comment]
Hi Rob!

Thanks for opening the issue. I've commited a new version of the Inference notebook which allows switching between CPU/GPU.
By default it uses a GPU if available, the CPU otherwise.

For the `AssertionError: Torch not compiled with CUDA enabled`, I've tested the notebook using the environment provided in the repo, and couldn't reproduce the issue on Windows 11.

To make sure I didn't install `cudatoolkit` manually, I tried a fresh install of the environment using the provided `environment.yml` file on Windows 11 using Miniforge. It installed `cudatoolkit` as well, but like you said, it's not an explicit requirement.

I've now added it as an explicit requirement, validated the updated YAML file and tested the notebook on it.

Might I trouble you to try reinstalling the environment now and see if it works, or to share your OS/environment so I can try reproducing the error?

Thanks again!
